src/flb_pack_gelf.c

```diff
@@ -671,16 +671,37 @@ flb_sds_t flb_msgpack_to_gelf(flb_sds_t *s, msgpack_object *o,
                     val = temp;
                     val_len = snprintf(temp, sizeof(temp) - 1,
                                        "%" PRIu64, v->via.u64);
+                    /*
+                     * Check if the value length is larger than our string.
+                     * this is needed to avoid stack-based overflows.
+                     */
+                    if (val_len > sizeof(temp)) {
```

Comment:
Indeed, seems legit that `val_len` could be larger than `sizeof(temp)` ...

```
RETURN VALUE
       Upon successful return, these functions return the number of characters
       printed (excluding the null byte used to end output to strings).

       The  functions  snprintf() and vsnprintf() do not write more than
       size bytes (including the terminating null byte ('\0')).  If the output
       was truncated due to this limit, then the re‐
       turn value is the number of characters (excluding the terminating
       null byte) which would have been written to the final string if enough
       space had  been  available.
```

Reply:
Yep! We have an input that triggers this exact scenario, i.e. where the val_len is set to 300+